README.md

```diff
@@ -97,6 +97,10 @@ model Post {
 - `@DtoEntityHidden` - omits field in `Entity`
 - `@DtoCreateOptional` - adds field **optionally** to `CreateDTO` - useful for fields that would otherwise be omitted (e.g. `@id`, `@updatedAt`)
 - `@DtoUpdateOptional` - adds field **optionally** to `UpdateDTO` - useful for fields that would otherwise be omitted (e.g. `@id`, `@updatedAt`)
+- @DtoCreateHidden - omits field in `CreateDTO`
+- @DtoUpdateHidden - omits field in `UpdateDTO`
+- `@DtoCreateApiResponse` - generate `@ApiResponseProperty()` and ignore all other annotations in `CreateDTO`
+- same for `DtoUpdateApiResponse` and `DtoPlainApiResponse`
```

Comment:
I don't get the point of `@ApiResponseProperty()`. Why should it be used over `@ApiProperty()`? ApiResponseProperty is the same just with fewer properties and `readOnly: true` ([api-property.decorator.ts#L75-L85](https://github.com/nestjs/swagger/blob/master/lib/decorators/api-property.decorator.ts#L75-L85)). For me, it does not make sense to use it in the CreateDto and UpdateDto (which are basically for request bodies) if the field is read-only. And in the plain Dto (the response body), there is no need to mark it as read-only. If you want to omit the field in Create or Update, there is the `@DtoReadOnly` annotation (or your new `@DtoCreateHidden` and `@DtoCreateHidden` for more fine control).

Or did I misunderstand something completely?

Reply:
this annotation do 2 things: makes prop readonly; remove all validation annotations(usecase: nestjs ValidationPipe.whitelist)
since I won't edit the generated files, it's convenient to reuse the dtos as response. it's ok to remove this annotation.